book/src/fork-selection.md

```diff
@@ -1,75 +1,115 @@
+* A voter can eventually recover from voting on a fork that doesn't become the
+  fork with the desired finality.
+* If the voters share a common ancestor then they will converge to a fork
+  containing that ancestor no matter how they are partitioned, although it may
+  not be the latest possible ancestor at the start of the fork.
+* Rollback requires exponentially more time for older votes than for newer
+  votes.
+* Voters have the freedom to set a minimum finality threshold before committing
+  a vote to a higher lockout.  This allows each voter to make a trade-off
+  between risk and reward. See [finality](#economic-finality).
 
 ## Time
 
-For networks like Solana, time can be the PoH hash count, which is a VDF that provides a source of time before consensus. Other networks adopting this approach would need to consider a global source of time.
+For networks like Solana, time can be the PoH hash count, which is a VDF that
+provides a source of time before consensus. Other networks adopting this
+approach would need to consider a global source of time.
 
-For Solana, time uniquely identifies a specific leader for fork generation.  At any given time only 1 leader, which can be computed from the ledger itself, can propose a fork.  For more details, see [fork generation](fork-generation.md) and [leader rotation](leader-rotation.md).
+For Solana, time uniquely identifies a specific leader for fork generation.  At
+any given time only 1 leader, which can be computed from the ledger itself, can
+propose a fork.  For more details, see [fork generation](fork-generation.md)
+and [leader rotation](leader-rotation.md).
 
 ## Algorithm
 
-The basic idea to this approach is to stack consensus votes.  Votes in the stack must be for forks that descend from each other and for forks that are valid in the ledger they are submitted to.  Each consensus vote has a `lockout` in units of time before it can be discarded.  When a vote is added to the stack, the lockouts of all the previous votes in the stack are doubled (see [Rollback](#Rollback)).  With each new vote, a voter commits the previous votes to an ever-increasing lockout.  Since at 32 votes we can consider the system to be at `max lockout` any votes with a lockout above `1<<32` are dequeued (FIFO).  Dequeuing a vote is the trigger for a reward.  If a vote expires before it is dequeued, it and all the votes above it are popped (LIFO) from the vote stack.  The voter needs to start rebuilding the stack from that point.
+The basic idea to this approach is to stack consensus votes.  Votes in the
+stack must be for forks that descend from each other and for forks that are
+valid in the ledger they are submitted to.  Each consensus vote has a `lockout`
+in units of time before it can be discarded.  When a vote is added to the
```

Comment:
what's the smallest lockout period?  one block?

Reply:
@rob-solana i had 2 blocks in simulation.  Not sure, maybe 2 slots?

Reply:
if a block is ~800ms, then max lockout is arrived at in ~25s, and lasts for ~100 years?

Reply:
@rob-solana we can make the curve whatever we want.  `C + (2^N)/K`